javascript/253-Meeting-Rooms-ii.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @param {number[][]} intervals
+ * Time O((N * logN) + (M * logM)) | Space O(1)
+ * @return {number}
+ */
+ var minMeetingRooms = function(intervals) {
+    const { start, end } = splitIntervals(intervals);
+    let [ minRooms, startIndex, endIndex ] = [ 0, 0, 0 ];
+
+    while (startIndex < intervals.length) {
+        const [ startTime, endTime ]  = [ start[startIndex], end[endIndex] ];
+
+        const hasGap = (prevEnd, currStart) =>
+            prevEnd <= currStart;
+
+        if (hasGap(endTime, startTime)) {
+            minRooms--;
+            endIndex++;
+        }
+
+        minRooms++;
+        startIndex++;
+    }
+
+    return minRooms;
+};
+
+const splitIntervals = (intervals, start = [], end = []) => {
+    for (const [ startTime, endTime ] of intervals) {
+        start.push(startTime);
+        end.push(endTime);
+    }
+
+    const sort = ((a, b) => a - b);
```

Comment:
nit: a better name is maybe `comparator` or `intervalComparator`. "sort" is a bit overloaded because it usually refers to the method that does the actual sorting